src/main/java/io/neonbee/internal/verticle/ServerVerticle.java

```diff
@@ -1,366 +1,283 @@
-            throw new UnsupportedOperationException("OAuth2 authentication handler is not implemented yet");
-        case "redirect":
-            // TODO return RedirectAuthHandler.create(authProvider, loginRedirectURL, returnURLParam);
-            throw new UnsupportedOperationException("Redirect authentication handler is not implemented yet");
-        default:
-            throw new IllegalArgumentException("Unknown authentication handler type in server configuration");
-        }
-    }
 
-    private AuthenticationProvider createAuthProvider(JsonObject options) {
-        if (LOGGER.isDebugEnabled()) {
-            LOGGER.debug("Configuring auth. provider with options: {}", options);
-        }
+            Endpoint endpoint;
+            try {
+                endpoint =
+                        Class.forName(endpointType).asSubclass(Endpoint.class).getDeclaredConstructor().newInstance();
+            } catch (ClassNotFoundException e) {
+                LOGGER.error("No class for endpoint type {}", endpointType, e);
+                return failedFuture(new IllegalArgumentException("Endpoint class not found", e));
+            } catch (ClassCastException e) {
+                LOGGER.error("Endpoint type {} must implement {}", endpointType, Endpoint.class.getName(), e);
+                return failedFuture(
+                        new IllegalArgumentException("Endpoint does not implement the Endpoint interface", e));
+            } catch (NoSuchMethodException e) {
+                LOGGER.error("Endpoint type {} must expose an empty constructor", endpointType, e);
+                return failedFuture(new IllegalArgumentException("Endpoint does not expose an empty constructor", e));
+            } catch (InvocationTargetException | IllegalAccessException | InstantiationException e) {
+                LOGGER.error("Endpoint type {} could not be instantiated or threw an exception", endpointType, e);
+                return failedFuture(Optional.of((Exception) e.getCause()).orElse(e));
+            }
 
-        switch (options.getString("type", EMPTY)) {
-        case "htdigest":
-            // TODO return HtdigestAuth.create(vertx, htfile);
-            throw new UnsupportedOperationException("HtDigest authentication provider is not implemented yet");
-        case "htpasswd":
-            // TODO return HtpasswdAuth.create(vertx, new HtpasswdAuthOptions(jsonConfig));
-            throw new UnsupportedOperationException("HtPasswd authentication provider is not implemented yet");
-        case "jdbc":
-            // TODO return JDBCAuth.create(vertx, client).setAllSettersHere;
-            throw new UnsupportedOperationException("JDBC authentication provider is not implemented yet");
-        case "jwt":
-            JsonArray pubSecKeys = options.getJsonArray("pubSecKeys");
-            return JWTAuth.create(vertx, new JWTAuthOptions().setPubSecKeys(extractPubSecKeys(pubSecKeys)));
```

Comment:
You have never replaced that, but it is necessary. Maybe it is fixed in Vert.x 4.0.3 but I don't think so.

Reply:
What do you mean with "that"? the extractPubSecKeys? I just replaced it with the respective method in Vert.x and I checked the implementation, it looked essentially exactly like the think you did in extractPubSecKeys, or am I mistaking?

Reply:
PreProcessor shoudl replace place holder with "BEGIN PRIVATE KEY\n" + "verificationkey\n" + "END PRIVATE KEY"